libs/vectorise/include/vectorise/fixed_point/fixed_point.hpp

```diff
@@ -1037,6 +659,2027 @@ FixedPoint<I, F> const FixedPoint<I, F>::CONST_MIN{FixedPoint::FromBase(FixedPoi
 template <std::uint16_t I, std::uint16_t F>
 FixedPoint<I, F> const FixedPoint<I, F>::NaN{
     FixedPoint::FromBase(FixedPoint::Type(1) << (FixedPoint::TOTAL_BITS - 1))};
+template <std::uint16_t I, std::uint16_t F>
+FixedPoint<I, F> const FixedPoint<I, F>::POSITIVE_INFINITY{
+    FixedPoint::NaN |
+    FixedPoint::FromBase(FixedPoint::Type(1) << (FixedPoint<I, F>::FRACTIONAL_BITS - 1))};
+template <std::uint16_t I, std::uint16_t F>
+FixedPoint<I, F> const FixedPoint<I, F>::NEGATIVE_INFINITY{
+    FixedPoint::NaN |
+    FixedPoint::FromBase(FixedPoint::Type(3) << (FixedPoint<I, F>::FRACTIONAL_BITS - 2))};
+
+template <std::uint16_t I, std::uint16_t F>
+std::ostream &operator<<(std::ostream &s, FixedPoint<I, F> const &n)
+{
+  std::ios_base::fmtflags f(s.flags());
+  s << std::setprecision(F / 4);
+  s << std::fixed;
+  if (FixedPoint<I, F>::isNaN(n))
+  {
+    s << "NaN";
+  }
+  else if (FixedPoint<I, F>::isPosInfinity(n))
+  {
+    s << "+∞";
+  }
+  else if (FixedPoint<I, F>::isNegInfinity(n))
+  {
+    s << "-∞";
+  }
+  else
+  {
+    s << double(n);
+  }
+  s << " (0x" << std::hex << n.Data() << ")";
+  s.flags(f);
+  return s;
+}
+
+////////////////////
+/// constructors ///
+////////////////////
+
+/**
+ * Templated constructor when T is an integer type
+ * @tparam T any integer type
+ * @param n integer value to set FixedPoint to
+ */
+template <std::uint16_t I, std::uint16_t F>
+template <typename T>
+constexpr FixedPoint<I, F>::FixedPoint(T n, meta::IfIsInteger<T> *)
+  : data_{static_cast<typename FixedPoint<I, F>::Type>(n)}
+{
+  assert(CheckNoOverflow(n));
+  Type s    = (data_ < 0) - (data_ > 0);
+  Type abs_ = s * data_;
+  abs_ <<= FRACTIONAL_BITS;
+  data_ = s * abs_;
+}
+
+/**
+ * Templated constructor only when T is a float/double type
+ * @tparam T any float/double type
+ * @param n float/double value to set FixedPoint to
+ */
+template <std::uint16_t I, std::uint16_t F>
+template <typename T>
+constexpr FixedPoint<I, F>::FixedPoint(T n, meta::IfIsFloat<T> *)
+  : data_(static_cast<typename FixedPoint<I, F>::Type>(n * ONE_MASK))
+{
+  assert(CheckNoOverflow(n * ONE_MASK));
+  //    assert(CheckNoRounding<T>());
```

Comment:
leftover code?

Reply:
Well, not sure about rounding. It's a fact that we will have rounding at least for the 2 FP types <16,16> and <32,32>, and not for <64,64> or bigger types. So the CheckNoRounding<T> feels somehow redundant at the moment.

Reply:
I will leave it there for now and will revisit CheckNoRounding when/if we tackle larger types like <64,64>